hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/ddl/TestSpark3DDL.scala

```diff
@@ -742,6 +744,8 @@ class TestSpark3DDL extends HoodieSparkSqlTestBase {
             option("hoodie.schema.on.read.enable","true").
             option("hoodie.datasource.write.reconcile.schema","true").
             option(DataSourceWriteOptions.TABLE_NAME.key(), tableName).
+            option(HoodieWriteConfig.WRITE_PAYLOAD_CLASS_NAME.key(), classOf[OverwriteWithLatestAvroPayload].getName).
```

Comment:
So these changes are made only to make the test pass?

Reply:
Yes

Reply:
Kind of think we should modify the test to make it pass by using new default payloads.

Reply:
This test was written for `OverwriteWithLatestAvroPayload` and it checks OverwriteWithLatest behavior, i think it's better to let it be. Or may be add the same test and make it pass by using new default payloads. But... 
All other tests use default payload and pass. Why it is not enough?

Reply:
> This test was written for OverwriteWithLatestAvroPayload and it checks OverwriteWithLatest behavior, i think it's better to let it be

It does not looks like so.

Reply:
> It does not looks like so.
Yes, you are right. This test is not about checking OverwriteWithLatest behavior, but `OverwriteWithLatestAvroPayload` made it work determinate in case of randomly generated input data.   
This test with HOODIE_AVRO_DEFAULT and random values in preCombine field failures, so i just set `ts` values manually on each write. Is it better?

Reply:
> It does not looks like so.
Yes, you are right. This test is not about checking OverwriteWithLatest behavior, but `OverwriteWithLatestAvroPayload` made it work determinate in case of randomly generated input data.   
This test with HOODIE_AVRO_DEFAULT and random values in preCombine field failures, so i just set `ts` values manually on each write. Is it better?

Reply:
The `#preCombine` of `OverwriteWithLatestAvroPayload` and `DefaultHoodieRecordPayload` should be the same because the inheritence. Are you saying the discrepency of `#combineAndGetUpdateValue`?

Reply:
> Are you saying the discrepency of #combineAndGetUpdateValue?
Exactly. `OverwriteWithLatestAvroPayload` overwrites storage with latest delta record, `DefaultHoodieRecordPayload` chooses the latest record based on ordering field value.  
So for this test to pass with `DefaultHoodieRecordPayload`  we must not use random generated values in ordering field (ts). Thats what i did in my last fix.